runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingPlugin.java

```diff
@@ -242,50 +239,58 @@ public void onInteractingChanged(InteractingChanged event)
 
 		currentSpot = spot;
 	}
+	@Getter
+	private Set<FishingTools> usableGear;
 
-	private boolean canPlayerFish(final ItemContainer itemContainer)
+	private boolean canPlayerFish(final ItemContainer... itemContainer)
 	{
-		if (itemContainer == null)
-		{
-			return false;
-		}
+		usableGear.clear();
+		for(ItemContainer container : itemContainer){
+			if (container == null)
+			{
+				break;
```

Comment:
Why is the loop exited if one of the passed containers is null? It should just skip that container, not all subsequent containers.